libs/core/langchain_core/output_parsers/json.py

```diff
@@ -194,11 +207,29 @@ class JsonOutputParser(BaseCumulativeTransformOutputParser[Any]):
     describing the difference between the previous and the current object.
     """
 
-    pydantic_object: Optional[Type[BaseModel]] = None
+    pydantic_object: Optional[Type[TBaseModel]] = None  # type: ignore
+    json_schema: Optional[Dict[str, Any]] = None
```

Comment:
Could we remove the new attribute json_schema? As an API it's not obvious what it's doing.

@jnis23 I'm also not sure how many of these changes we should be making. We'll be dropping support for pydantic 1 + pydantic 2 v1 in about 3 months, and most of the conditional behavior is going to make the refactor in 3 months more complicated to execute.

Reply:
The thought behind `json_schema` was that if users had their JSON schema already defined, they could just `json.load` the schema definition file and pass the dict to the parser instead of having to recreate the schema as a pydantic object. Agreed there is probably a better name and would also need some doc. I'll remove for now and it can be added as a new feature if someone has that use case.

Sounds good on no more changes, the full upgrade to v2 will clean this up a lot. I'll remove `json_schema` and push this back up. Feel free to close this if we want to hold off until the full v2 upgrade

Reply:
Because json_schema and pydantic_object are used for the same purpose (defining a schema), it makes more sense to rename `pydantic_object` to schema and have it accept a union of pydantic or JSON schema.

In this case I don't think it's worth making a change -- it's just changing the format instructions associated with this parser which is used for llms / chat models that don't support tool calling. In that case it's unclear that JSONSchema is actually a good schema to use (it's too verbose). (https://github.com/eyurtsev/kor likely works better with typescript schema)